### DISCUSSION TITLE
HTML Rendering

### DISCUSSION THREAD

**Discussion by JPhilC** [StereoKit/StereoKit#211] [Ideas]
Hi 

I need to render HTML content onto a slate. In Unity I currently use a 3rd party web view control in my VS/Unity app. Is anyone working on some form of webview control? I am primarily interested for the Hololens 2 and its use in industrial applications.

Phil

**Comment by maluoi** [comment]
The easiest way will probably always be to use something like a LaunchURI command! This could be a little tricky based on the samples I've peeked at, but will likely be easier than any alternative. HoloLens will then display a browser slate inside your application as built-in OS functionality. One big advantage here is that this provides a _real_ web rendering experience.

That's a fairly simple solution, but may not cut it for every circumstance.

There's currently no StereoKit specific web renderers in development, but if you can find one that outputs a color buffer, then it should not be so difficult to load that into a Texture for display in StereoKit. I don't have any specific suggestions in mind as I haven't tackled this problem myself yet, but it seems that there are a number of solutions in this category at various levels of quality and platform support.

**Reply by JPhilC** [reply]
Thanks for the reply. I'm currently looking into the cefSharp project. On the face of it there is no support for UWP or Arm64 but wondering how big a job it would be to compile the sources for these targets, if that is even possible. 

BTW, I'm amazed you have time to turn out StereoKit and your other projects as well as organising the MR events etc. Keep up the good work!

**Reply by JPhilC** [reply]
Since my last reply (literally minutes ago) someone from the CEF project has responded and apparently CEF won't compile for C++/WinRT. They mentioned that WebView2 might be an option so I'll look into whether that can be used offscreen to generate a bitmap.
